packages/hooks/src/useDocumentEventListener.ts

```diff
@@ -0,0 +1,37 @@
+import { useEffect, useRef } from 'react';
+
+/**
+ * Hook to subscribe to an event listener.
+ * @param type Represents the event type to listen for.
+ * @param eventCallback Event listener callback function.
+ * @param enabled Determines whether or not the hook should run, defaults to true.
+ * @param options Optional parameter to specify options passed to the eventListener.
+ * @param dependencies Optional array to be passed to useEffect hook, such that the hook will only run if the array's values have changed.
+ */
+export default function useDocumentEventListener(
+  type: string,
+  eventCallback: (e) => void,
+  options?: object,
+  dependencies?: Array<any>,
+  enabled: boolean = true,
+) {
+  const memoizedEventCallback = useRef(e => {}); //eslint-disable-line @typescript-eslint/no-unused-vars
+
+  useEffect(() => {
+    memoizedEventCallback.current = eventCallback;
+  }, [eventCallback]);
+
+  useEffect(() => {
+    if (!enabled) {
+      return;
+    }
+
+    const callback = e => memoizedEventCallback.current(e);
+
+    document.addEventListener(type, callback, options);
+
+    return () => {
+      document.removeEventListener(type, eventCallback);
+    };
+  }, [enabled, ...(dependencies ? dependencies : [])]);
```

Comment:
Thinking about this, we might want
```js
...(dependencies ? dependencies : [])
```
to be
```js
...(dependencies ? dependencies : [type, eventCallback, options])
```

Also, see Harry's note in the comment on the type for `dependencies` as the conversation feels relevant.

Reply:
If it is present, is it not fair to expect an array? Not defaulting to an empty array in the absence of the presence of the arg

Reply:
I actually don't think that we should spread enabled into the dependencies. I get that it's intuitive that you'd want to listen for changes to that value, but I think that if we're letting users pass in their own array of dependencies, we shouldn't add anything to potentially break that. (Speaking from experience, this broke ours). I'd be open to including it as a default in dependencies, however, if a user passes nothing in for dependencies.